paimon-trino-common/src/main/java/org/apache/paimon/trino/TrinoFilterConverter.java

```diff
@@ -69,6 +72,7 @@
 public class TrinoFilterConverter {
 
     private static final Logger LOG = LoggerFactory.getLogger(TrinoFilterConverter.class);
+    public static final Pattern IS_NESTED = Pattern.compile(".+\\[.+]");
```

Comment:
Please refer to https://github.com/apache/paimon/pull/3221, it does not use regex anymore